lib/WTSI/NPG/HTS/PacBio/Sequel/AnalysisPublisher.pm

```diff
@@ -120,6 +136,11 @@ sub publish_sequence_files {
             $self->find_pacbio_runs($self->_metadata->run_name,
                                     $self->_metadata->well_name,
                                     $self->_get_tag_name_from_fname($filename));
+
+        if (@tag_records != 1) {
+          $self->logcroak("Unexpected barcode from $file for SMRT cell ",
+              $self->_metadata->well_name, ' run ', $self->_metadata->run_name);
```

Comment:
Is it possible that `@tag_records` doesn't contain exactly 1 item because it's empty? If so, is the error message still accurate? (I don't know enough to tell - just that an empty array seems possible).

Reply:
In that block we've found a BAM file to load for a specific tag. We are erroring if the data for that run, cell, tag is not found in the mlwarehouse. In previous versions of SMRT Link the deplexing was done differently so we had to allow archiving of unexpected tags but now we assume unexpected tags in the deplexing are an error in SMRT Link or the mlwarehouse (SequenceScape or TRACTION) which should be resolved before loading.

Reply:
I see - the "unexpected" barcode belongs to the file we are working with and if a barcode gives zero or >1 result, then there's something wrong (either with the barcode, or the warehouse, but we're assuming the warehouse isn't wrong). I still think it's a _little_ bit ambiguous (as we can have the expected barcode while warehouse is wrong and get the error), but not enough to block this.

Reply:
Yes we do have that situation where the mlwarehouse is wrong. We hope if will not to get that far though as it should fail the pre QC check  (if deplexed we expect at least a certain amount of the data deplexed). In larger pools it's always hard to police automatically though.